hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -307,7 +307,10 @@ public void refreshTimeline() throws IOException {
           if (!commitMetadata.getMetadata(CHECKPOINT_KEY).isEmpty()) {
             resumeCheckpointStr = Option.of(commitMetadata.getMetadata(CHECKPOINT_KEY));
           }
-        }  else if (HoodieTimeline.compareTimestamps(HoodieTimeline.FULL_BOOTSTRAP_INSTANT_TS,
+        } else if (commitMetadata.getOperationType() == WriteOperationType.CLUSTER) {
+          // incase of CLUSTER commit, no checkpoint will be available in metadata.
```

Comment:
@nsivabalan This should be fine since the commit timeline to read the CHECKPOINT from does not include the clustering instants.

Reply:
`This should be fine since the commit timeline to read the CHECKPOINT from does not include the clustering instants.`

@n3nash @nsivabalan This is not correct.  The commit time does have the clustering instants (r`eplacecommit`) based on the changes in #2048.  So this breaks the logic of getting the last checkpoint, even if we have the walk-back logic in #4034, which is skipped.